### PR TITLE
cephadm/box: create osds with podman.

### PR DESCRIPTION
**Pull request by pereman2**
Based on @zmc 's changes on https://github.com/zmc/ceph/tree/wip-box-rootless-podman
this changes introduce the creation of OSDs on podman by creating
loopback devices.

For this to work there are severeral things necessary:
* mount /sys/dev/block
* --privileged to mount /sys/dev/block fs. CAP SYS_ADMIN is not enough.
* mount /sys/block
* mount /sys/devices/virtual/block. In conjunction with mounts above
  this abilitates lslbk to work which is needed by ceph-volume.
* created OSD loopback images are saved on loop-images by default
* devices.json provides information of loopback devices currently used

Furthermore, this PR removes podman-compsoe in favor of a simpler approach
to reduce complexity. Podman alone will be used to deploy containers.

**how to test**:
```sh
./box.py -v cluster setup
./box.py -v cluster start # simple cluster without extra hosts and osds, box_hosts_0 will be the main container
./box.py -v cluster down
./box.py -v cluster start --expanded --hosts 2 # this will create two extra hosts (box_hosts_1, and box_hosts_2).
```

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by jmolmo — CHANGES_REQUESTED** [review]
Do we have documentation for box utilization?

**Review by epuertat — COMMENTED** [review]
Nice to see ongoing work with this! Thanks @pereman2 !

File: src/cephadm/box/util.py [hunk 1]
```diff
@@ -174,37 +207,47 @@ def engine():
...
     return f'{engine()}-compose'
 
+def get_seed_name():
+    if engine() == 'docker':
+        return 'seed'
+    elif engine() == 'podman':
+        return 'box_hosts_0'
+    else:
+        print(f'unkown engine {engine()}')
+        sys.exit(1)
```
[the remark this answers is not in the document]
Comment (pereman2) [reply src/cephadm/box/util.py:204]: done

**Comment by pereman2** [comment]
> Do we have documentation for box utilization?

Yes we do, https://docs.ceph.com/en/latest/dev/cephadm/developing-cephadm/#cephadm-box-container-podman-inside-podman-development-environment

**Review by nizamial09 — APPROVED** [review]
worked for me! thank you

**Comment by pereman2** [comment]
jenkins test make check

**Comment by pereman2** [comment]
jenkins test api

**Review by jmolmo — CHANGES_REQUESTED** [review]
Error testing from scratch (no loop devices defined), or when you try to create OSDs backed by loop devices

```
  File "/home/jolmomar/Code/ceph/src/cephadm/box/osd.py", line 48, in create_loopback_device
    dev_minor = re.match(r'\/dev\/[^\d]+(\d+)', loop_dev).groups()[0]
                ^^
NameError: name 're' is not defined
```

Maybe it would be a good idea to have a command  for cleaning OSDs, for example:
if you execute 
# losetup -a
/dev/loop2: [0038]:3263123 (/home/jolmomar/Code/ceph/src/cephadm/box/loop-images/osd2
you have the information about the things need to be cleaned,

So it would be nice to have a box command also to delete loopd devices used by osds; for example:
./box.py osd destroy

And also is probably a good idea to include the osd cleanup in the cluster cleanup command.

-----------------------------------------------------------------------------------------------------------
More things....
-  Not clear at all how to interact with the cluster once the cluster has been deployed ( the dashboard url provided does not work)
-  How can i execute ceph commands in the box container?
-  Do we have documentation of this tool?

If the user decides to add more services once the cluster is created , and these services  require to use a external port in the container, how is the port opened? Is this feature something supported?

**Comment by pereman2** [comment]
@jmolmo you can check the docs in [this link](https://docs.ceph.com/en/latest/dev/cephadm/developing-cephadm/#cephadm-box-container-podman-inside-podman-development-environment). Cleanup was already implemented with `box.py cluster cleanup` but I've included `box.py osd destroy`.

If you want to run cephadm commands you can enter either host with the usual `podman exec -it box_hosts_<num> bash` or `./box.py -v cluster bash` to bash inside the seed container(container where bootstrap was run). After going inside bash `cephadm shell` is enough to run mgr commands.

**Review by jmolmo — APPROVED** [review]

**Pull request merged**